### PR TITLE
HOT-2050 Prevent participants from disappearing from screening

### PR DESCRIPTION
**Pull request by wmitchell-cambria**
### Jira Story

- [Screening Payload: Include New People Added HOT-2050](https://osi-cwds.atlassian.net/browse/HOT-2050)
- The task above is a subtask, so the state of this is more accurately being tracked on [HOT-2018](https://osi-cwds.atlassian.net/browse/HOT-2018) (As I'm writing this, the high today is 100 degrees, so that ticket number seems apt.)

## Description

Prior to this PR, we tracked participants in two locations in Redux: the `participants` list, and on the screening as `screening.participants`. The latter is not kept up to date as participants are posted and deleted, and should not be used. So, this PR does two things:
1. Provides selectors for Participants, so that everything can consistently select from the same place in Redux.
2. Removes the participants from the screenings stored in Redux, so there is no opportunity to use the wrong one.

**_!!! This is important !!!_** Because Ferb will soon DELETE participants that are not listed when making PUT requests that update the screening, there is also now a very dangerous chance of data loss if we even once update a screening without specifying the most up to date participant list. To avoid this, we check for common errors in the sagas that update screenings and abort if necessary.

## Tests
- [x] I have included unit tests 
- [ ] I have included feature tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

## Types of changes

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] I have ran all tests for the project. *(I ran them once partway through...will need them to run again now, though.)*
- [x] I have ran lint/rubocop check for the changed/new files.
- [x] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build, to check Lint/Rubocop, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

**Comment by wmitchell-cambria** [comment]
Jenkins, retest this please.

**Comment by wmitchell-cambria** [comment]
Jenkins, retest this please.

**Comment by wmitchell-cambria** [comment]
> Why are we only checking for an undefined participants list? That seems like weird edge case which we should be handling by setting participants to an empty list by default.

It is a weird edge case, yes. At first, I wasn't going to guard against it at all, and indeed it should never happen in our code base today. All of the selectors that are used to update participants merge in the correct participants list and default to an empty array if necessary. However, because the mistake of not merging that list is so easy, combined with the drastic side effects of posting no participants to the API, I wanted to include the check to protect us from this in future changes. Unit tests and feature tests didn't feel sufficient to cover the scenario.

> Also, why are we not storing participants in the screening? They are an integral part of the screening logically -- so doesn't it make more sense to keep them there? Is it because of snapshot?

A couple reasons. Generally, we were already treating the separate participants list as our source of truth, so the choice was between normalizing to that one for the few cases that were improperly using the other list or changing that source of truth to the nested list on the screening. Snapshot was a factor in this, as it does not have a screening object to nest within, and keeping them separate also (loosely) follows the guidelines on [normalizing Redux state shape](https://redux.js.org/recipes/structuring-reducers/normalizing-state-shape).

> Also, we need to make sure that we can still save appropriate cards when there are no participants. Do we have a test that checks that edge case?

Yup. Individual selectors used to get the screening with edits for each card have unit tests to ensure that they default to an empty participant list when none are present, the `saveScreeningCardSaga` tests to ensure it handles updates when the participants list is empty, and most card-specific feature tests like `spec/features/screening/screening_information_spec.rb` verify screening edits with no participants present.

**Comment by TylerOSI** [comment]
Thank you for the comments -- they help clarify some things. However, It seems like we are guarding against a problem that should be handled by the API. If we send a list of participants as 'null' then the API should throw an error because participants is a list. They should only be accepting a list -- either an empty one or one with participants.

Please check current behavior of Ferb. If its accepting participants as null then we have a problem. If not, then we shouldn't need this check for undefined.

File: app/javascript/snapshots/SnapshotSideBar.jsx [hunk 1]
```diff
@@ -10,7 +10,7 @@ const SnapshotSideBar = (props) => (
         <NavLink key={1} text='People & Roles' href='#search-card-anchor' >
           <NavLinks nested={true}>
             <div className='nested-block'>
-              {props.participants && props.participants.map(({id, first_name, last_name, name_suffix}) =>
+              {props.participants.map(({id, first_name, last_name, name_suffix}) =>
```
Comment (wmitchell-cambria) [inline app/javascript/snapshots/SnapshotSideBar.jsx]: Vikas asked about this offline, so some clarification... This check was unnecessary before, as the proptypes marked the participants prop as required _and_ there was a default value of an empty array just in case. However, the real reason I was confident in removing it is that all of these components now fetch participants through the `selectParticipants` selector, which will provide a default List if none is present. We can be very confident that we will have a valid array here. (Same goes for `ScreeningSideBar`.)

**Comment by wmitchell-cambria** [comment]
retest this please.
